drivers/misc/ipts/companion/ipts-surface.c

```diff
@@ -49,11 +74,11 @@ static int ipts_surface_probe(struct platform_device *pdev)
 		return -ENODEV;
 	}
 
-	ret = intel_ipts_add_fw_handler(&ipts_surface_request_firmware,
-		(void *)acpi_device_hid(adev));
+	ipts_surface_companion.data = (void *)acpi_device_hid(adev);
+	ret = ipts_add_companion(&ipts_surface_companion);
 	if (ret) {
-		dev_info(&pdev->dev, "Adding IPTS firmware handler failed, "
-			"error: %d\n", ret);
+		dev_info(&pdev->dev, "Adding IPTS companion failed, "
```

Comment:
why is this info and not error?

Reply:
Because it could happen that multiple companions get probed, or the user explicitly tells IPTS to ignore the companion. In that case, we didn't really want the companion to error.

Reply:
I think multiple companions probing should be at least a warning, if not an error. I'd normally expect that only one companion driver gets loaded for a specific machine. Also, as far as I can tell, if a user sets the modparam to ignore the IPTS companion, `ipts_add_companion` does not return an error, so I think it's safe to change this to something like `dev_warn(..., "IPTS companion driver already present, aborting\n")`.

Reply:
Done, changed `dev_info` to `dev_warn`.